### DISCUSSION TITLE
Feature Request and Questions about Open-Meteo

### DISCUSSION THREAD

**Discussion by lauguimel** [open-meteo/open-meteo#1162] [Q&A]
Hello Patrick,

First of all, thank you for sharing this excellent project with the community. It is truly a valuable resource, and I greatly appreciate your work.

I have developed an application that uses various NWP models (such as ICON-EU, GFS, Arome, etc.) and am considering migrating to Open-Meteo for its flexibility and accessibility. However, I have a few specific needs and questions that I hope you can assist with:

1. I specifically require some additional variables that are not currently available, such as **HPBL** (planetary boundary layer height). While this parameter is present in the raw models (called MH in ICON-EU and HCOULIM in Arome), it does not seem to be available via Open-Meteo. Would it be possible to include these variables by opening an issue or through another process?

2. Regarding upper-level values: is there a way to get output for levels in **meters** (e.g., Arome provides 24 levels between 10 and 3000 meters)? This would be incredibly helpful for certain use cases.

3. It appears that GRIB downloads are not directly supported in Open-Meteo. However, I noticed that it is possible to transform `om` files into NetCDF using the export function. Is there an efficient way to merge all `om` files before this step? This would simplify workflows, such as generating tiled precipitation images.

4. From what I understand, temperature correction is applied based on altitude, using a lapse rate of **0.7°C per 100 meters**. However, this adjustment is far from accurate in certain conditions, such as during winter in mountainous regions with inversions or stratus, or in summer under anticyclonic conditions with high stability. Is it possible to disable this correction to retain the raw model output?

Your support and insight on these questions would be greatly appreciated. Thank you again for creating and maintaining this amazing tool!

Best regards,

**Comment by patrick-zippenfenig** [comment]
Hi,

1. `Boundary Layer Height PBL` is available for [GFS and HRRR models](https://open-meteo.com/en/docs/gfs-api#latitude=46.8757&longitude=8.6654&hourly=temperature_2m,boundary_layer_height). I might consider adding it for ICON and AROME as well. Can you open an issue ticket for it?

2. We do not process all model and height levels. Only pressure level variables are relatively consistent across different models/providers. Adding more or all height levels would require large amounts of storage and increase operational burden to copy all data to different nodes while only <1% users are interested in upper level data. That being said, I would like to add "all". A potential approach would be a dedicated API endpoint exclusively for upper level data that uses fewer nodes, but with more storage (~4-8 TB) per node. Maybe Q2 next year I have a look at this.

3. We are actively working on the next `om`-file format revision which will provide Python/TS/Rust/Swift libraries to read `om`-files directly. See https://github.com/open-meteo/open-meteo/pull/1018. No ETA yet possible.

4. Yes, you can add `&elevation=NaN` which will then use the model elevation.
